flink-ml-tests/src/test/java/org/apache/flink/test/iteration/BoundedAllRoundCheckpointITCase.java

```diff
@@ -131,16 +129,7 @@ static JobGraph createVariableAndConstantJobGraph(
             int maxRound,
             int failoverCount,
             SinkFunction<OutputRecord<Integer>> sinkFunction) {
-        StreamExecutionEnvironment env =
-                StreamExecutionEnvironment.getExecutionEnvironment(
-                        new Configuration() {
-                            {
-                                this.set(
-                                        ExecutionCheckpointingOptions
-                                                .ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH,
-                                        true);
-                            }
-                        });
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         env.getConfig().enableObjectReuse();
```

Comment:
Would it be better if used `TestUtils.getExecutionEnvironment` here to disable kryo for these tests?

Reply:
The purpose of FLINK-30688 is to ensure that all algorithms can be run without using kryo. And this can be verified by having all tests in flink-ml-lib use `TestUtils.getExecutionEnvironment` to disable kryo.

Now that all tests in flink-ml-lib (except OnlineLogisticRegression) have used `TestUtils.getExecutionEnvironment` to disable kryo, it seems that we don't have to use `TestUtils.getExecutionEnvironment ` for those tests in `flink-ml-iteration`. So I think it is simpler to keep those tests in `flink-ml-iteration` as is.

I removed the config `ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH` here because `ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH` is enabled by default since Flink 1.15.